pkg/storage/mvcc.go

```diff
@@ -990,7 +990,7 @@ func mvccGetMetadata(
 		if !hasPoint || !unsafeKey.Key.Equal(metaKey.Key) {
 			meta.Deleted = true
 			meta.Timestamp = rkTimestamp.ToLegacyTimestamp()
-			return true, int64(EncodedMVCCKeyPrefixLength(metaKey.Key)), 0, hlc.Timestamp{}, nil
+			return true, 0, 0, hlc.Timestamp{}, nil
```

Comment:
We don't use it, but after reading comments for the Nth time I realized that this value is meant to represent extra data that needs to be deleted if all keys go. In case of bare range, there's nothing to delete. We do skip the whole key if we detect that realKeyChanged is empty and meta is not inlined so we don't need to look at it or use it.